vlasovsolver/cpu_acc_transform.cpp

```diff
@@ -98,53 +109,329 @@ Eigen::Transform<Real,3,Eigen::Affine> compute_acceleration_transformation(
    const Real rhoq = spatial_cell->parameters[CellParams::RHOQ_V] + EPSILON;
    const Real hallRhoq =  (rhoq <= Parameters::hallMinimumRhoq ) ? Parameters::hallMinimumRhoq : rhoq ;
    const Real hallPrefactor = 1.0 / (physicalconstants::MU_0 * hallRhoq );
 
+   // Bulk velocity is used to transform to a frame where the motional E-field vanishes
    Eigen::Matrix<Real,3,1> bulk_velocity(spatial_cell->parameters[CellParams::VX_V],
                                          spatial_cell->parameters[CellParams::VY_V],
                                          spatial_cell->parameters[CellParams::VZ_V]);
 
-   // compute total transformation
-   Transform<Real,3,Affine> total_transform(Matrix<Real, 4, 4>::Identity()); //CONTINUE
+   // Use electron solvers for anything below half a proton mass
+   bool smallparticle = false;
+   Eigen::Matrix<Real,3,1> electronV(0.,0.,0.);
+   if (getObjectWrapper().particleSpecies[popID].mass < 0.5*physicalconstants::MASS_PROTON) {
+      smallparticle = true;
+      // Store the original electron bulk velocity
+      electronV[0] = spatial_cell->get_population(popID).V_V[0];
+      electronV[1] = spatial_cell->get_population(popID).V_V[1];
+      electronV[2] = spatial_cell->get_population(popID).V_V[2];
+      /*
+      if ( (spatial_cell->parameters[CellParams::CELLID]>1408) && (spatial_cell->parameters[CellParams::CELLID]<1410) ) {
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " electron V_V 0 " << spatial_cell->get_population(popID).V_V[0] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " electron V_V 1 " << spatial_cell->get_population(popID).V_V[1] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " electron V_V 2 " << spatial_cell->get_population(popID).V_V[2] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " EJE 0 " << spatial_cell->parameters[CellParams::EXJE] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " EJE 1 " << spatial_cell->parameters[CellParams::EYJE] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " EJE 2 " << spatial_cell->parameters[CellParams::EZJE] << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 0 " << Bx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 1 " << By << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " B 2 " << Bz << endl;
+
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBXdy " << dBXdy << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBXdz " << dBXdz << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBYdx " << dBYdx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBYdz " << dBYdz << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBZdx " << dBZdx << endl;
+	std::cerr << spatial_cell->parameters[CellParams::CELLID] << " dBZdy " << dBZdy << endl;
+	}*/
```

Comment:
This looks like extremely ad-hoc debugging code so it can be dropped.

Reply:
Will be dropped when everything works! :)